pkg/endpoint/providers/azure/kubernetes/client.go

```diff
@@ -21,18 +21,18 @@ const (
 )
 
 // NewClient creates the Kubernetes client, which retrieves the AzureResource CRD and Services resources.
-func NewClient(kubeClient kubernetes.Interface, azureResourceKubeConfig *rest.Config, Controller k8s.Controller, stop chan struct{}, cfg configurator.Configurator) (*Client, error) {
+func NewClient(kubeClient kubernetes.Interface, azureResourceKubeConfig *rest.Config, kubeController k8s.KubeController, stop chan struct{}, cfg configurator.Configurator) (*Client, error) {
 	azureResourceClient := osmClient.NewForConfigOrDie(azureResourceKubeConfig)
 
-	k8sClient := newClient(kubeClient, azureResourceClient, Controller)
+	k8sClient := newClient(kubeClient, azureResourceClient, kubeController)
 	if err := k8sClient.Run(stop); err != nil {
 		return nil, errors.Errorf("Failed to start %s client: %+v", kubernetesClientName, err)
 	}
 	return k8sClient, nil
 }
 
 // newClient creates a provider based on a Kubernetes client instance.
-func newClient(kubeClient kubernetes.Interface, azureResourceClient *osmClient.Clientset, kubeController k8s.Controller) *Client {
+func newClient(kubeClient kubernetes.Interface, azureResourceClient *osmClient.Clientset, kubeController k8s.KubeController) *Client {
```

Comment:
I was thinking its okay for the interface to be `k8s.Controller`, but this is okay too.

Reply:
I think you are right. the Kube part for the interface name is redundant as there is the package prefix in most situations already. I think the I "_replaced all_" that. will change back to controller.